application/libraries/Omeka/Storage/Adapter/ZendS3.php

```diff
@@ -63,6 +64,10 @@ public function __construct(array $options = array())
         if (!empty($options[self::ENDPOINT_OPTION])) {
             $this->_s3->setEndpoint($options[self::ENDPOINT_OPTION]);
         }
+        
+        if (isset($this->_options[self::FORCE_SSL]) && parse_url($this->_s3->getEndpoint(), PHP_URL_SCHEME)=='http'){
+            $this->_options['FORCE_SSL_DISPLAY_ENDPOINT'] = 'https:' . substr($this->_s3->getEndpoint(), 5);
```

Comment:
This should probably be a property on the class and not in `_options`, it's clearer if `_options` is just the options set by the user only.

Reply:
Do you have a preference regarding the endpoint logic? e.g.

```
$endpoint = isset($this->_force_ssl_display_endpoint) ? $this->_force_ssl_display_endpoint : $this->_s3->getEndpoint();
```

versus...

`$endpoint = somePrivateMethod()`;